frontend/src/components/admin/EditTextModal.tsx

```diff
@@ -0,0 +1,75 @@
+import { Flex, VStack, Image } from "@chakra-ui/react";
+import React, { useContext, useState } from "react";
+import {
+  EditorChangeStatuses,
+  EditorContextType,
+  ModuleEditorParams,
+
+  ContentBlock,
+
+} from "../../types/ModuleEditorTypes";
+import { TextInput } from "../common/TextInput";
+import { Modal } from "../common/Modal";
+
+import { TextArea } from "../common/TextArea";
+
+export interface EditTextModalProps {
+  onClose: () => void;
+  isOpen: boolean;
+  block: ContentBlock;
+  index: number;
+  context: EditorContextType;
+}
+
+const EditTextModal = ({
+  block,
+  index,
+  isOpen,
+  onClose,
+  context,
+}:
+EditTextModalProps): React.ReactElement => {
+  const [text, setText] = useState(block.content.text ?? "");
+
+  if (!context) return <></>;
+  const { state, dispatch } = context;
+
+  const onSave = () => {
+    const newBlock = {
+      ...block,
+      content: {
+        ...block.content,
+        text,
+      },
+    };
+
+    dispatch({
+      type: "update-block",
+      value: {
+        index,
+        block: newBlock,
+      },
+    });
+  
+    onClose();
+  };
+
+  return (
+    // eslint-disable-next-line react/jsx-props-no-spreading
+    <Modal
+      size="xl"
+      header="Edit Text Component"
+      onConfirm={onSave}
+      onCancel={onClose}
+      isOpen={isOpen}
+    >
+      <Flex>
+        <VStack flex="1">
+          <TextArea defaultValue={text} onChange={setText} isRequired />
```

Comment:
Unfortunately, just setting `isRequired` doesn't actually validate that the field contains text. Would you be able to add some validation in `onSave`? You can set the `errorMessage` prop if there's an error.

Also, it would be good for accessibility to add a label to this field (pass a `label` prop).